src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1013,10 +1013,7 @@ public SocketAsyncContext(SafeCloseSocket socket)
 
         private void Register()
         {
-            // Note, on OSX, this is not always true because in certain cases, 
-            // the socket can already be in non-blocking mode even though we didn't set that ourselves.
-            // TODO: Track down exactly why this is
-            // Debug.Assert(_nonBlockingSet);
+            Debug.Assert(_nonBlockingSet);
```

Comment:
I understand the changes made in the native code, but how does that affect the value of _nonBlockingSet in the managed code?

Reply:
This code gets executed when we get EAGAIN from a socket operation like send() or recv().

When this happens, we expect we've previously set the socket to non-blocking.  That's what the assert is testing.

Reply:
Ok.  But if this assert was previously failing, that means we were hitting this code path due to such an EAGAIN from send/recv.  With the changes made in the .c file, we will still hit this code path for an EAGAIN, right?  What in the .c change will either case _nonBlockingSet to be true now when it previously would have been false, or will change us not to follow this code path when it is false?  I'm obviously missing something obvious :)

Reply:
The changes in the .c file should ensure the socket is not in non-blocking mode unless we explicitly set it to non-blocking mode (at which time we will set _nonBlockingSet).

So we should not receive EAGAIN unless we've explicitly set the socket to non-blocking mode.

Reply:
Aha.  Got it.  So it's the "will change us not to follow this code path when it is false" case.  Thanks.